modules/core/05-port/types/module.go

```diff
@@ -104,6 +104,15 @@ type IBCModule interface {
 		packet channeltypes.Packet,
 		relayer sdk.AccAddress,
 	) error
+
+	// SetICS4Wrapper sets the ICS4Wrapper. This function may be used after
+	// the module's initialization to set the middleware which is above this
+	// module in the IBC application stack.
+	// The ICS4Wrapper **must** be used for sending packets and writing acknowledgements
+	// to ensure that the middleware can intercept and process these calls.
+	// Do not use the channel keeper directly to send packets or write acknowledgements
+	// as this will bypass the middleware.
```

Comment:
Just so I understand: this has always been a requirement of middlewares (to not use the channel keeper directly)?

Reply:
Not sure where we are on docs with this, but we should:
1. Make sure the docs are clear about this
2. Link to that in the migration guide